### DISCUSSION TITLE
Two inputs controlled by the same fieldPath

### DISCUSSION THREAD

**Discussion by vittau** [react-hook-form/react-hook-form#6814] [Q&A] [answered]
Hello everyone,

My use case requires me to have two different controlled inputs (I'm using `Controller`) that update their values simultaneously, regardless of which one you type in.
Is there any way to do this, that is not `watch`ing for the value and calling `setValue` in an `useEffect`? Performance really suffers by doing this...

Thanks!

**Comment by bluebill1049** [comment, marked as answer]
if `watch` hit performance issue, then you should consider to use `useWatch` so the re-render is isolated.

**Reply by vittau** [reply]
I figured that wouldn't help much because I'm calling `setValue` anyway, but turns out it still helped quite a lot. I suppose `setValue` doesn't really change any of the other props of the form, so it still avoids re-renders.
Thank you!
